core/state.py

```diff
@@ -1991,6 +1992,17 @@ def SetValue(self, lval, val, which_scopes, flags=0):
                             strs[lval.index] = rval.s
                         return
 
+                    elif case2(value_e.SparseArray):
+                        lhs_sp = cast(value.SparseArray, UP_cell_val)
+                        error_code = bash_impl.SparseArray_SetElement(
+                            lhs_sp, mops.BigInt(lval.index), rval.s)
```

Comment:
```suggestion
                            lhs_sp, mops.BigInt(lval.index), rval.s)
```

When should we use `mops.BigInt(int)` and when `mops.IntWiden(int)`?

Reply:
I received a reply in Zulip: https://oilshell.zulipchat.com/#narrow/channel/121539-oil-dev/topic/mops.2EBigInt.28i.29.20vs.2E.20mops.2EIntWiden.28i.29/near/485491386